more_itertools/more.py

```diff
@@ -3592,19 +3593,21 @@ def _sample_weighted(iterable, k, weights):
             weights_to_skip -= weight
 
     # Equivalent to [element for weight_key, element in sorted(reservoir)]
+    shuffle(reservoir)
     return [heappop(reservoir)[1] for _ in range(k)]
 
 
 def sample(iterable, k, weights=None):
     """Return a *k*-length list of elements chosen (without replacement)
-    from the *iterable*. Like :func:`random.sample`, but works on iterables
-    of unknown length.
+    from the *iterable*. Similar to :func:`random.sample`, but works on
```

Comment:
Consider documenting that if the input iterable is sized k or smaller, `more_itertools.sample()` will return all of the input in shuffled order, possibly giving a result length smaller than *k*.

Alternatively, consider adding a `strict` option to raise an exception as `random.sample()` does. That would let the user decide whether it is an error to say, "choose 100 distinct notes on an 88-key piano."